airflow/contrib/kubernetes/pod_launcher.py

```diff
@@ -74,15 +74,25 @@ def run_pod(self, pod, startup_timeout=120):
                 time.sleep(1)
             self.log.debug('Pod not yet started')
 
-        final_status = self._monitor_pod(pod)
+        final_status = self._monitor_pod(pod, get_logs)
         return final_status
 
-    def _monitor_pod(self, pod):
+    def _monitor_pod(self, pod, get_logs):
         # type: (Pod) -> State
 
-        while self.pod_is_running(pod):
-            self.log.info("Pod {} has state {}".format(pod.name, State.RUNNING))
-            time.sleep(2)
+        if get_logs:
+            logs = self._client.read_namespaced_pod_log(
```

Comment:
Is it okay that this is synchronously done here? The async variant lets us pass a callback IIRC.

Reply:
So my thoughts here is that airflow operators run synchronously. They either own a thread entirely (like in the LocalExecutor, SequentialExecutor), or they completely own the machine (i.e. KubernetesExecutor, CeleryExecutor). We also want to block here because we're waiting for the pod to succeed or fail before we return the final status to the operator.

Reply:
Is it a use-case for people to try and check scheduler logs during the course of the job running? My understanding here is that the scheduler doesn't get receive any of the logs till the job completes (succeeds/fails) - so it would remain blank until the logs appear all at once. Is that correct?

Reply:
TBH I'm not sure, but even if you're not getting the logs, it's still just entering an infinite sleep loop until the pod completes, so either way it accomplishes the same goal of holding the process until the pod has terminated. Are there any potential issues that could arise from this though?